### ISSUE TITLE
Strange Train-Accuracy at very beginning

### ISSUE THREAD

**Issue by luoyetx** [tornadomeet/ResNet#19]
I got a very strange train-accuracy output like below. 

```
INFO:root:Namespace(aug_level=2, batch_size=128, bn_mom=0.9, data_dir='data', data_type='imagenet', depth=34, gpus='0,1', kv_store='device', list_dir='data', lr=0.1, model_load_epoch=0, mom=0.9, num_examples=1281167, retrain=False, wd=0.0001)
[10:13:13] src/io/iter_image_recordio.cc:209: ImageRecordIOParser: data/train_480_q90.rec, use 4 threads for decoding..
[10:13:14] src/io/iter_image_recordio.cc:209: ImageRecordIOParser: data/val_256_q90.rec, use 4 threads for decoding..
INFO:root:Start training with [gpu(0), gpu(1)]
INFO:root:Epoch[0] Batch [50]   Speed: 323.01 samples/sec       Train-accuracy=0.507188
INFO:root:Epoch[0] Batch [50]   Speed: 323.01 samples/sec       Train-top_k_accuracy_5=0.932031
INFO:root:Epoch[0] Batch [100]  Speed: 308.74 samples/sec       Train-accuracy=0.026719
INFO:root:Epoch[0] Batch [100]  Speed: 308.74 samples/sec       Train-top_k_accuracy_5=0.501719
INFO:root:Epoch[0] Batch [150]  Speed: 313.60 samples/sec       Train-accuracy=0.060937
INFO:root:Epoch[0] Batch [150]  Speed: 313.60 samples/sec       Train-top_k_accuracy_5=0.265313
INFO:root:Epoch[0] Batch [200]  Speed: 311.01 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [200]  Speed: 311.01 samples/sec       Train-top_k_accuracy_5=0.105938
INFO:root:Epoch[0] Batch [250]  Speed: 313.49 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [250]  Speed: 313.49 samples/sec       Train-top_k_accuracy_5=0.070938
INFO:root:Epoch[0] Batch [300]  Speed: 312.41 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [300]  Speed: 312.41 samples/sec       Train-top_k_accuracy_5=0.084687
INFO:root:Epoch[0] Batch [350]  Speed: 312.28 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [350]  Speed: 312.28 samples/sec       Train-top_k_accuracy_5=0.084844
INFO:root:Epoch[0] Batch [400]  Speed: 313.47 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [400]  Speed: 313.47 samples/sec       Train-top_k_accuracy_5=0.079531
INFO:root:Epoch[0] Batch [450]  Speed: 312.25 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [450]  Speed: 312.25 samples/sec       Train-top_k_accuracy_5=0.045937
INFO:root:Epoch[0] Batch [500]  Speed: 313.55 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [500]  Speed: 313.55 samples/sec       Train-top_k_accuracy_5=0.074531
INFO:root:Epoch[0] Batch [550]  Speed: 312.60 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [550]  Speed: 312.60 samples/sec       Train-top_k_accuracy_5=0.062344
INFO:root:Epoch[0] Batch [600]  Speed: 313.40 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [600]  Speed: 313.40 samples/sec       Train-top_k_accuracy_5=0.052969
INFO:root:Epoch[0] Batch [650]  Speed: 312.94 samples/sec       Train-accuracy=0.000000
INFO:root:Epoch[0] Batch [650]  Speed: 312.94 samples/sec       Train-top_k_accuracy_5=0.048281
```

The accuracy is very high and goes down very quickly. It seems my training data doesn't get shuffled. I use `im2rec data/train.lst /home/zhangjie/data/imagenet/ILSVRC2012/train/ data/train_480_q90.rec resize=480 quality=90` to create the data.

**Comment by tornadomeet** [comment]
it should due to the rec file is not correct.  train.lst should shuffle begin using im2rec. does the same classes in your `train.lst`  is in adjacent? @luoyetx

**Comment by luoyetx** [comment]
@tornadomeet this is part of the `train.lst`

```
0 0 n01440764/n01440764_10026.JPEG
1 0 n01440764/n01440764_10027.JPEG
2 0 n01440764/n01440764_10029.JPEG
3 0 n01440764/n01440764_10040.JPEG
4 0 n01440764/n01440764_10042.JPEG
5 0 n01440764/n01440764_10043.JPEG
6 0 n01440764/n01440764_10048.JPEG
7 0 n01440764/n01440764_10066.JPEG
8 0 n01440764/n01440764_10074.JPEG
9 0 n01440764/n01440764_1009.JPEG
10 0 n01440764/n01440764_10095.JPEG
11 0 n01440764/n01440764_10108.JPEG
12 0 n01440764/n01440764_10110.JPEG
13 0 n01440764/n01440764_10120.JPEG
14 0 n01440764/n01440764_10124.JPEG
15 0 n01440764/n01440764_10150.JPEG
16 0 n01440764/n01440764_10159.JPEG
17 0 n01440764/n01440764_10162.JPEG
18 0 n01440764/n01440764_10183.JPEG
19 0 n01440764/n01440764_10194.JPEG
20 0 n01440764/n01440764_10211.JPEG
21 0 n01440764/n01440764_10218.JPEG
22 0 n01440764/n01440764_10227.JPEG
23 0 n01440764/n01440764_1024.JPEG
24 0 n01440764/n01440764_10251.JPEG
25 0 n01440764/n01440764_10254.JPEG
26 0 n01440764/n01440764_10281.JPEG
27 0 n01440764/n01440764_10290.JPEG
28 0 n01440764/n01440764_10293.JPEG
29 0 n01440764/n01440764_10307.JPEG
30 0 n01440764/n01440764_10317.JPEG
31 0 n01440764/n01440764_10319.JPEG
32 0 n01440764/n01440764_10320.JPEG
33 0 n01440764/n01440764_10323.JPEG
```

You mean I should shuffle this file by myself before using `im2rec`?

**Comment by luoyetx** [comment]
I find the doc about `im2rec` [here](http://mxnet.readthedocs.io/en/latest/packages/python/io.html). It doesn't shuffle the items in `lst` file. I should shuffle this file myself. Too bad, I need to re-generate the `rec` file. :(

**Comment by tornadomeet** [comment]
@luoyetx  yes, we need to shuffle it first. because the `rec` is  continuous save on disk for quick read, so if not shuffled, when we feed data to train, it will always train the same category, and this will break the convergence.  
sorry for the inconvenient, i'll add it to the readme.

BTW, you can use the train.lst from http://data.dmlc.ml/mxnet/models/imagenet/resnet/, then the classification id is the same with our pre-trained model. (it has been shuffled, and you can shuffled it again.)

**Comment by tornadomeet** [comment]
yes, generate the rec is time costy! but we only need to generate it once, and can use it for any other model. :)

**Comment by luoyetx** [comment]
@tornadomeet Thanks for providing the `lst` file. I will try this files.
